File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    
    stack = []
    for i in s:
        if i == '(':
            stack.append(i)
        elif i == ')':
            if stack and stack[-1] == '(':
                stack.pop()
            else:
                return False
    
    if len(stack) == 0:
        return True
    else:
        return False

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    var stack = [];
    
    for (b of s) {
        if (stack.length === 0) {
            if (b === ")") return false
            stack.push(b);
        } else {
            if (b === "(") stack.push(b);
            else {
                if (stack[stack.length-1] === ')') return false
                else stack.pop()
            }
        }
    }
    if (stack.length) return false;
    
    return answer;
}